packages/ember-glimmer/tests/integration/helpers/unbound-test.js

```diff
@@ -0,0 +1,600 @@
+
+    this.assertText('No spans here, son.');
+
+    this.runTask(() => set(this.context, 'content', {
+      anUnboundString: 'No spans here, son.'
+    }));
+
+    this.assertText('No spans here, son.');
+  }
+
+  ['@test should be able to use unbound helper in #each helper']() {
+    this.render(`<ul>{{#each items as |item|}}<li>{{unbound item}}</li>{{/each}}</ul>`, {
+      items: emberA(['a', 'b', 'c', 1, 2, 3])
+    });
+
+    this.assertText('abc123');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('abc123');
+  }
+
+  ['@test should be able to use unbound helper in #each helper (with objects)']() {
+    this.render(`<ul>{{#each items as |item|}}<li>{{unbound item.wham}}</li>{{/each}}</ul>`, {
+      items: emberA([{ wham: 'bam' }, { wham: 1 }])
+    });
+
+    this.assertText('bam1');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('bam1');
+
+    this.runTask(() => this.context.items.setEach('wham', 'HEY'));
+
+    this.assertText('bam1');
+
+    this.runTask(() => set(this.context, 'items',  emberA([{ wham: 'bam' }, { wham: 1 }])));
+
+    this.assertText('bam1');
+  }
+
+  ['@test it should assert unbound cannot be called with multiple arguments']() {
+    let willThrow = () => {
+      this.render(`{{unbound foo bar}}`, {
+        foo: 'BORK',
+        bar: 'BLOOP'
+      });
+    };
+
+    expectAssertion(willThrow, /unbound helper cannot be called with multiple params or hash params/);
+  }
+
+  ['@test should render on attributes']() {
+    this.render(`<a href="{{unbound foo}}"></a>`, {
+      foo: 'BORK'
+    });
+
+    this.assertHTML('<a href="BORK"></a>');
+
+    this.runTask(() => this.rerender());
+
+    this.assertHTML('<a href="BORK"></a>');
+
+    this.runTask(() => set(this.context, 'foo', 'OOF'));
+
+    this.assertHTML('<a href="BORK"></a>');
+
+    this.runTask(() => set(this.context, 'foo', 'BORK'));
+
+    this.assertHTML('<a href="BORK"></a>');
+  }
+
+  ['@htmlbars syntax form updates on parent re-render']() {
+    let fooBarInstance;
+    let FooBarComponent = Component.extend({
+      init() {
+        this._super(...arguments);
+        fooBarInstance = this;
+      },
+      foo: 'BORK',
+      barBinding: 'foo'
+    });
+
+    this.registerComponent('foo-bar', {
+      ComponentClass: FooBarComponent,
+      template: `{{unbound foo}} {{unbound bar}}`
+    });
+
+    this.render(`{{foo-bar}}`);
+
+    this.assertText('BORK BORK');
+
+    this.runTask(() => this.rerender());
+
+    this.assertText('BORK BORK');
+
+    this.runTask(() => {
+      set(fooBarInstance, 'foo', 'OOF');
+      this.rerender();
+    });
+
+    this.assertText('OOF OOF');
+
+    this.runTask(() => set(fooBarInstance, 'foo', 'BLIP'));
+
+    this.assertText('OOF OOF');
+
+    this.runTask(() => {
+      set(fooBarInstance, 'foo', 'BORK');
+      this.rerender();
+    });
+
+    this.assertText('BORK BORK');
+  }
+
+  ['@test sexpr form does not update on parent re-render']() {
```

Comment:
Can you put in a comment saying the semantics here is not guaranteed? The [original commit](https://github.com/emberjs/ember.js/commit/508d9de45bd6d729c8cb0954f97fd7ed83531276) that introduced the test actually asserted the opposite, but we later [flipped the test](https://github.com/emberjs/ember.js/pull/11965) when it started failing. I think the intention was that we didn't care at the time so it doesn't really matter what it does (we should have deleted or skipped that test at the time, imo) and didn't really intend to lock down the inconsistent semantics.

I *think* the `does not update unless explicitly told to` is both more consistent and probably easier for us to implement.